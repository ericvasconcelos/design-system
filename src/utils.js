export { arrayCombinations } from 'utils/arrayCombinations'
export { getBackgroundColor } from 'utils/getBackgroundColor'
export { getBorder } from 'utils/getBorder'
export { getButtonStyle } from 'utils/getButtonStyle'
export { getBorderRadius } from 'utils/getBorderRadius'
export { getColorFromName } from 'utils/getColorFromName'
export { getGuttersNames } from 'utils/getGuttersNames'
export { getIconNames } from 'utils/getIconNames'
export { getPadding } from 'utils/getPadding'
export { getPaddingNames } from 'utils/getPaddingNames'
export { getTypography } from 'utils/getTypography'
export { parsePalleteName } from 'utils/parsePalleteName'
export { styleColorValidator } from 'utils/styleColorValidator'
export { useAriaProps } from 'utils/useAriaProps'

export { DateUtils } from 'utils/DateUtils'
export { QueryUtils } from 'utils/QueryUtils'
export { ObjectUtils }  from 'utils/ObjectUtils'
export { CookiesUtils } from 'utils/CookiesUtils'
export { MoneyUtils } from 'utils/MoneyUtils'
export { NumberUtils } from 'utils/NumberUtils'