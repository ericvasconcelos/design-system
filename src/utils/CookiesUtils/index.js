export class CookiesUtils {
  static toObject(string = '') {
    const cookies = {}
    if (string !== '') {
      const parts = string.split('; ')
      parts.forEach(part => {
        const [key, value] = part.split('=')
        cookies[key] = value
      })
    }
    return cookies
  }
}
