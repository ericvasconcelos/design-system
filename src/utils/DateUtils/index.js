export class DateUtils {
  static toDate(string, format = 'pt-br') {
    let day = ''
    let month = ''
    let year = ''

    switch (format) {
      case 'pt-br':
        [day, month, year] = string?.split('/')
        break
      case 'iso':
        [year, month, day] = string?.split('-')
        break
      default:
        throw new Error('Format invalid')
    }

    return new Date(year, parseInt(month, 10) - 1, day)
  }

  static getAge(date) {
    if (!(date instanceof Date)) {
      throw new TypeError('Date invalid')
    }

    const difference = Date.now() - date.getTime()

    return Math.abs(new Date(difference).getUTCFullYear() - 1970)
  }

  static toString(date, format = 'pt-br') {
    if (!(date instanceof Date)) {
      throw new TypeError('Date invalid')
    }

    const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate()
    const month = date.getMonth() + 1 < 10 ? `0${date.getMonth() + 1}` : date.getMonth() + 1
    const year = date.getFullYear()

    switch (format) {
      case 'pt-br':
        return `${day}/${month}/${year}`
      case 'iso':
        return `${year}-${month}-${day}`
      default:
        throw new Error('Format invalid')
    }
  }

  static validate(date, format = 'pt-br') {
    const monthDaysMap = {
      '01': 31,
      '02': 29,
      '03': 31,
      '04': 30,
      '05': 31,
      '06': 30,
      '07': 31,
      '08': 31,
      '09': 30,
      10: 31,
      11: 30,
      12: 31
    }
    let isValid = false

    let day = ''
    let month = ''
    let year = ''

    switch (format) {
      case 'pt-br':
        [day, month, year] = date?.split('/')
        break
      case 'iso':
        [year, month, day] = date?.split('-')
        break
      default:
        throw new Error('Format invalid')
    }

    isValid = isValid || parseInt(day, 10) <= monthDaysMap[month]
    isValid = isValid && parseInt(year, 10) > 1900

    return isValid
  }

  static convertMonthToYear(month) {
    return parseInt(month, 10) / 12
  }

  static formatPeriod(period, periodType) {
    const periodTypeMap = {
      monthly: { singular: 'mÃªs', plural: 'meses' },
      yearly: { singular: 'ano', plural: 'anos' },
      daily: { singular: 'dia', plural: 'dias' }
    }
    return `${period} ${periodTypeMap[periodType][period < 2 ? 'singular' : 'plural']}`
  }
}
