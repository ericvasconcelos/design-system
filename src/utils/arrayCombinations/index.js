export const arrayCombinations = (...args) => {
  const returnable = []
  const max = args.length - 1
  function helper(arr, i) {
    for (let j = 0, { length } = args[i]; j < length; j++) {
      const newArray = arr.slice(0) // clone arr
      newArray.push(args[i][j])
      if (i === max) returnable.push(newArray)
      else helper(newArray, i + 1)
    }
  }
  helper([], 0)
  return returnable
}
