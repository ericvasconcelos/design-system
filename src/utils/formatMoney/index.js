import { toMoney, toNumber } from 'vanilla-masker'

export const formatMoney = entry => {
  const moneyFormatter = {
    precision: 2,
    separator: ',',
    delimiter: '.',
    unit: 'R$'
  }

  return toMoney(toNumber(entry), moneyFormatter)
}
