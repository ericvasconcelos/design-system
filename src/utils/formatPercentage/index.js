import { toMoney, toNumber } from 'vanilla-masker'

export const formatPercentage = entry => {
  const percentageFormatter = {
    precision: 1,
    separator: ',',
    suffixUnit: '%'
  }
  return toMoney(toNumber(entry), percentageFormatter)
}
