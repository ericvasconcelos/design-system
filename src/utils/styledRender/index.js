import React from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { theme } from 'styles/theme'

export const styledRender = (children, options = {}) => {
  const rendered = render(<ThemeProvider theme={theme}>{children}</ThemeProvider>, options)
  return {
    ...rendered,
    rerender: (ui, rerenderOptions = {}) =>
      styledRender(ui, { container: rendered.container, ...rerenderOptions })
  }
}
