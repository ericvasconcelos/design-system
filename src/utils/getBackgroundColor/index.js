import { getColorFromName } from 'utils/getColorFromName'

export const getBackgroundColor = ({ theme, styleColor }) =>
  `background-color: ${getColorFromName({ theme, styleColor })};`
