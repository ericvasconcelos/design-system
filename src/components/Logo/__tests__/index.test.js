import React from 'react'

import { styledRender } from 'utils/styledRender'

import { Logo } from '../index.js'

describe('Logo component render when', () => {
  it('default render', () => {
    const { getByTestId } = styledRender(<Logo />)
    const component = getByTestId('Logo')

    expect(component).toHaveAttribute(
      'src',
      'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'
    )
    expect(component).toHaveAttribute('title', 'Logo CashMe azul claro')
    expect(component).toHaveAttribute('alt', 'Logo CashMe azul claro')
    expect(component).toMatchSnapshot()
  })

  it('white no text render', () => {
    const { getByTestId } = styledRender(<Logo logoType="white" />)
    const component = getByTestId('Logo')

    expect(component).toHaveAttribute(
      'src',
      'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'
    )
    expect(component).toHaveAttribute('title', 'Logo CashMe branco sem texto')
    expect(component).toHaveAttribute('alt', 'Logo CashMe branco sem texto')
    expect(component).toMatchSnapshot()
  })

  it('white with text render', () => {
    const { getByTestId } = styledRender(<Logo logoType="whiteText" />)
    const component = getByTestId('Logo')

    expect(component).toHaveAttribute(
      'src',
      'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'
    )
    expect(component).toHaveAttribute('title', 'Logo CashMe branco')
    expect(component).toHaveAttribute('alt', 'Logo CashMe branco')
    expect(component).toMatchSnapshot()
  })

  it('darkBlue no text render', () => {
    const { getByTestId } = styledRender(<Logo logoType="darkBlue" />)
    const component = getByTestId('Logo')

    expect(component).toHaveAttribute(
      'src',
      'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'
    )
    expect(component).toHaveAttribute('title', 'Logo CashMe azul escuro sem texto')
    expect(component).toHaveAttribute('alt', 'Logo CashMe azul escuro sem texto')
    expect(component).toMatchSnapshot()
  })

  it('darkBlue with text render', () => {
    const { getByTestId } = styledRender(<Logo logoType="darkBlueText" />)
    const component = getByTestId('Logo')

    expect(component).toHaveAttribute(
      'src',
      'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'
    )
    expect(component).toHaveAttribute('title', 'Logo CashMe azul escuro')
    expect(component).toHaveAttribute('alt', 'Logo CashMe azul escuro')
    expect(component).toMatchSnapshot()
  })

  it('lightBlue no text render', () => {
    const { getByTestId } = styledRender(<Logo logoType="lightBlue" />)
    const component = getByTestId('Logo')

    expect(component).toHaveAttribute(
      'src',
      'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'
    )
    expect(component).toHaveAttribute('title', 'Logo CashMe azul claro sem texto')
    expect(component).toHaveAttribute('alt', 'Logo CashMe azul claro sem texto')
    expect(component).toMatchSnapshot()
  })

  it('lightBlue with text render', () => {
    const { getByTestId } = styledRender(<Logo logoType="lightBlueText" />)
    const component = getByTestId('Logo')

    expect(component).toHaveAttribute(
      'src',
      'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'
    )
    expect(component).toHaveAttribute('title', 'Logo CashMe azul claro')
    expect(component).toHaveAttribute('alt', 'Logo CashMe azul claro')
    expect(component).toMatchSnapshot()
  })

  it('Logo with alt and title custom render', () => {
    const { getByTestId } = styledRender(
      <Logo logoType="lightBlueText" alt="Alt text" title="Title text" />
    )
    const component = getByTestId('Logo')

    expect(component).toHaveAttribute(
      'src',
      'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'
    )
    expect(component).toHaveAttribute('title', 'Title text')
    expect(component).toHaveAttribute('alt', 'Alt text')
    expect(component).toMatchSnapshot()
  })
})
