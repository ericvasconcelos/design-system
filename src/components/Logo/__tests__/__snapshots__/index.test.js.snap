// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component render when Logo with alt and title custom render 1`] = `
.c0 {
  padding: 0;
}

<img
  alt="Alt text"
  class="c0"
  data-testid="Logo"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  title="Title text"
/>
`;

exports[`Logo component render when darkBlue no text render 1`] = `
.c0 {
  padding: 0;
}

<img
  alt="Logo CashMe azul escuro sem texto"
  class="c0"
  data-testid="Logo"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  title="Logo CashMe azul escuro sem texto"
/>
`;

exports[`Logo component render when darkBlue with text render 1`] = `
.c0 {
  padding: 0;
}

<img
  alt="Logo CashMe azul escuro"
  class="c0"
  data-testid="Logo"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  title="Logo CashMe azul escuro"
/>
`;

exports[`Logo component render when default render 1`] = `
.c0 {
  padding: 0;
}

<img
  alt="Logo CashMe azul claro"
  class="c0"
  data-testid="Logo"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  title="Logo CashMe azul claro"
/>
`;

exports[`Logo component render when lightBlue no text render 1`] = `
.c0 {
  padding: 0;
}

<img
  alt="Logo CashMe azul claro sem texto"
  class="c0"
  data-testid="Logo"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  title="Logo CashMe azul claro sem texto"
/>
`;

exports[`Logo component render when lightBlue with text render 1`] = `
.c0 {
  padding: 0;
}

<img
  alt="Logo CashMe azul claro"
  class="c0"
  data-testid="Logo"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  title="Logo CashMe azul claro"
/>
`;

exports[`Logo component render when white no text render 1`] = `
.c0 {
  padding: 0;
}

<img
  alt="Logo CashMe branco sem texto"
  class="c0"
  data-testid="Logo"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  title="Logo CashMe branco sem texto"
/>
`;

exports[`Logo component render when white with text render 1`] = `
.c0 {
  padding: 0;
}

<img
  alt="Logo CashMe branco"
  class="c0"
  data-testid="Logo"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  title="Logo CashMe branco"
/>
`;
